client/src/api/v2/camera.js

```diff
@@ -165,3 +165,40 @@ export function useCameraRecordingStatus(device) {
 export function getPrettyDeviceName(device) {
   return device === 'primary' ? 'Primary' : 'Secondary';
 }
+
+/**
+ * Initiate receiving camera video feed statuses
+ */
+function initVideoFeedStatus() {
+  emit('send-last-received-camera-video-feed-payloads');
+}
+
+/**
+ * @typedef {object} VideoFeedStatus
+ * @property {boolean}  online Whether video feed is on/off
+ */
+
+/**
+ * Returns the last received status of the video feeds from `/v3/camera/video-feed/status/<primary/secondary>`
+ *
+ * @returns {object.<string, VideoFeedStatus>} A VideoFeedStatus for each device
+ */
+export function useVideoFeedStatus() {
+  // Only run init once per render
+  useEffect(() => {
+    initVideoFeedStatus();
+  }, []);
+
+  const [payloads, setPayloads] = useState({}); // Payloads is Object.<string, string payload>
+
+  const handler = useCallback((newPayloads) => {
+    setPayloads(newPayloads);
+  }, []);
+
+  useChannel(`camera-video-feed-status`, handler);
+
+  return Object.keys(payloads).reduce((acc, key) => {
```

Comment:
This step could be skipped (see below). 


(I really don't like [reduce](https://www.youtube.com/watch?v=qaGjS7-qWzg) lol)

Reply:
I think stashing the `payload` object is a good idea. Why don't you like `reduce` @angustrau ? The body of the accumulator is the same as the body of the corresponding for loop so the semantics and syntax is mostly the same.

Reply:
Sorry in advance for another soapbox. From a FIT2102 perspective, reduce is a fine tool for functional programming, but some of the practices in the unit focus on correctness rather than readability or practicality. In an imperative code base like ours, there are often better, more specialized tools built in to the platform (such as `.find` for example). From the standpoint of an imperative programmer, the semantics are more unfamiliar than the equivalent for-loop, and require a bit more thought to process and understand what it is doing. In addition, the JavaScript syntax for reduce isn't great: the default value is after the reducer function, requiring you to skip to the end of the line then back to the start, interrupting the normal top-to-bottom flow when reading code. There is a place for reduce, especially when working with immutable and functional workflows (such as Redux), but there are better tools for the job here.

Reply:
> Sorry in advance for another soapbox. From a FIT2102 perspective, reduce is a fine tool for functional programming, but some of the practices in the unit focus on correctness rather than readability or practicality. In an imperative code base like ours, there are often better, more specialized tools built in to the platform (such as `.find` for example). From the standpoint of an imperative programmer, the semantics are more unfamiliar than the equivalent for-loop, and require a bit more thought to process and understand what it is doing. In addition, the JavaScript syntax for reduce isn't great: the default value is after the reducer function, requiring you to skip to the end of the line then back to the start, interrupting the normal top-to-bottom flow when reading code. There is a place for reduce, especially when working with immutable and functional workflows (such as Redux), but there are better tools for the job here.

I'd like to add on this why you barely see any complicated python list comprehension stuff in our repos. It may reduce number of lines and look cool 😎  but it actually makes it so much harder to read and understand initially. In the world of software engineering, writing readable and maintainable code will certainly be appreciated by your team mates.

Reply:
Done
5312dd9